Cubical/Categories/Adjoint.agda

```diff
@@ -0,0 +1,296 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Categories.Adjoint where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Data.Sigma
+open import Cubical.Categories.Category
+open import Cubical.Categories.Functor
+open import Cubical.Categories.NaturalTransformation
+open import Cubical.Foundations.Isomorphism
+
+open Functor
+
+open Iso
+open Precategory
+
+{-
+==============================================
+                  Overview
+==============================================
+
+This module contains two definitions for adjoint
+functors, and functions witnessing their
+logical (and maybe eventually actual?)
+equivalence.
+-}
+
+private
+  variable
+    ℓC ℓC' ℓD ℓD' : Level
+
+-- I tried using this... but Agda can never infer the category,
+-- even when it's pretty clear from context.....
+-- _⋆_ : ∀ {ℓ ℓ'} {C : Precategory ℓ ℓ'} {x y z} (f : C [ x , y ]) (g : C [ y , z ]) → C [ x , z ]
+-- _⋆_ {C = C} = seq C
+
+{-
+==============================================
+             Adjoint definitions
+==============================================
+
+We provide two alternative definitions for
+adjoint functors: the unit-counit
+definition, followed by the natural bijection
+definition.
+-}
+
+-- Adjoint def 1: unit-counit
+record _⊣_ {C : Precategory ℓC ℓC'} {D : Precategory ℓD ℓD'} (F : Functor C D) (G : Functor D C)
+                 : Type (ℓ-max (ℓ-max ℓC ℓC') (ℓ-max ℓD ℓD')) where
+  field
+    -- unit
+    η : 𝟙⟨ C ⟩ ⇒ (funcComp G F)
+    -- counit
+    ε : (funcComp F G) ⇒ 𝟙⟨ D ⟩
+    -- triangle identities
+    Δ₁ : PathP (λ i → NatTrans (F-lUnit {F = F} i) (F-rUnit {F = F} i))
```

Comment:
Do you always have to give these implicit arguments? Then they should be explicit

Reply:
Same for the other triangle

Reply:
Made explicit